docs/shared/reference/glossary.md

```diff
@@ -116,12 +136,24 @@ A repository with multiple [projects](#project).
 
 A [project](#project) that is located in a sub-folder of another project. This was made possible in Nx 15.3.
 
+### Nx Agents
+
+A paid service that hosts and manages customizable worker machines for use with [distributed task execution](#distributed-task-execution).
```

Comment:
At its essence, Nx Agents is a service to offshoring/externalizing your CI compute to Nx Cloud to make it more performant at lower cost. You do not necessarily need to use DTE for that, but it does DTE for you automatically when possible.